op_crates/fetch/lib.rs

```diff
@@ -96,6 +96,21 @@ pub fn init(isolate: &mut JsRuntime) {
   }
 }
 
+pub type SendRequestFuture =
+  Pin<Box<dyn Future<Output = Result<reqwest::Response, AnyError>>>>;
+
+pub trait SendRequest {
+  fn send(
+    &mut self,
+    _client: reqwest::Client,
+    builder: reqwest::RequestBuilder,
+  ) -> SendRequestFuture {
+    Box::pin(async move { builder.send().await.map_err(|err| err.into()) })
+  }
```

Comment:
This default implementation doesn't use the `reqwest::Client` but it comes in handy when modifying the request post facto:
```rust
let mut req = builder.build()?;
req.uri_mut().set_ip_host(ip)?;
let res = client.execute(req).await?;
Ok(res)
```